pyomo/core/base/sets.py

```diff
@@ -110,6 +110,36 @@ def wrapper_function ( *args, **kwargs ):
     return wrapper_function
 
 
+class _robust_sorter(object):
```

Comment:
I understand what this is doing after speaking with John. However, this class needs some more documentation to make it clear to others later.